lib/src/main/java/tech/relaycorp/relaydroid/ThirdPartyEndpoint.kt

```diff
@@ -1,5 +1,109 @@
 package tech.relaycorp.relaydroid
 
-public sealed class ThirdPartyEndpoint(override val address: String) : Endpoint
-public class PrivateThirdPartyEndpoint(address: String) : ThirdPartyEndpoint(address)
-public class PublicThirdPartyEndpoint(address: String) : ThirdPartyEndpoint(address)
+import org.bouncycastle.asn1.x500.style.BCStyle
+import tech.relaycorp.relaydroid.storage.persistence.PersistenceException
+import tech.relaycorp.relaynet.RelaynetException
+import tech.relaycorp.relaynet.issueDeliveryAuthorization
+import tech.relaycorp.relaynet.wrappers.x509.Certificate
+import tech.relaycorp.relaynet.wrappers.x509.CertificateException
+import java.security.PublicKey
+import java.time.ZonedDateTime
+
+public sealed class ThirdPartyEndpoint(
+    override val thirdPartyAddress: String
+) : Endpoint {
+
+    public companion object {
+
+        // Private
+
+        @Throws(PersistenceException::class)
+        public suspend fun loadPrivate(
```

Comment:
I'd suggest to move this method to `PrivateThirdPartyEndpoint.load()` -- I think that'd be more consistent with OOP inheritance where parents aren't aware of/responsible for children.

The only sort of exception to this would be methods that return an instance of a child, like `.load(...)` below.